controllers/suite_util_test.go

```diff
@@ -166,39 +157,16 @@ func getSampleComponentData(componentKey types.NamespacedName) *appstudiov1alpha
 // createComponentAndProcessBuildRequest create a component with specified build request and
 // waits until the request annotation is removed, which means the request was processed by the operator.
 // Use createCustomComponentWithBuildRequest if there is no need to wait.
-func createComponentAndProcessBuildRequest(componentKey types.NamespacedName, buildRequest string) {
-	createCustomComponentWithBuildRequest(componentConfig{componentKey: componentKey}, buildRequest)
-	waitComponentAnnotationGone(componentKey, BuildRequestAnnotationName)
-}
-
-// createComponentWithBuildRequest create a component with specified build request.
-// This method also sets devfile model, so the component is ready to be processed.
-func createComponentWithBuildRequest(componentKey types.NamespacedName, buildRequest string) {
-	createCustomComponentWithBuildRequest(componentConfig{componentKey: componentKey}, buildRequest)
-}
-
-// createComponentWithBuildRequestAndGitcreate a component with specified build request, and provided gitURL and revision.
-// This method also sets devfile model, so the component is ready to be processed.
-func createComponentWithBuildRequestAndGit(componentKey types.NamespacedName, buildRequest string, gitURL string, gitRevision string) {
-	createCustomComponentWithBuildRequest(componentConfig{componentKey: componentKey, gitURL: gitURL, gitRevision: gitRevision}, buildRequest)
-}
-
-func createCustomComponentWithoutBuildRequestWithoutDevfile(config componentConfig) {
-	component := getComponentData(config)
-	if component.Annotations == nil {
-		component.Annotations = make(map[string]string)
-	}
-
-	Expect(k8sClient.Create(ctx, component)).Should(Succeed())
-	getComponent(config.componentKey)
+func createComponentAndProcessBuildRequest(config componentConfig, buildRequest string) {
+	createCustomComponentWithBuildRequest(config, buildRequest)
+	waitComponentAnnotationGone(config.componentKey, BuildRequestAnnotationName)
 }
 
-func createCustomComponentWithBuildRequestWithoutDevfile(config componentConfig, buildRequest string) {
+func createCustomComponentWithoutBuildRequest(config componentConfig) {
```

Comment:
I'd name it just `createCustomComponent`

Reply:
well there is also "createComponentCustom"

also from the name is more clear what it is